private-templates-service/client/src/AuthenticatedComponent.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+import { UserType } from "./App";
+import Welcome from "./components/Welcome/Welcome";
+
+interface State {
+}
+interface AuthProps {
+  isAuthenticated: boolean,
+  user?: UserType,
+  authButtonMethod: () => Promise<void>
```

Comment:
Does this need to pass all the way through? I think the <Welcome /> component itself should be able to handle this logic

Reply:
From my understanding the props are passed to the requireAuthentication function, which propagates the props to the Dashboard/Welcome component. If I remove AuthProps, I'm not sure how to pull the props from the Component itself

Reply:
I meant specifically the authButtonMethod - can the authentication logic be handled by the class that owns the button itself?

Reply:
the login and logout functions were written in the App.tsx file, this authButtonMethod prop is used by Welcome and NavBar to define whether the user needs to log in/out. 

App.tsx: 
` <NavBar
            isAuthenticated={this.state.isAuthenticated}
            authButtonMethod={
              this.state.isAuthenticated
                ? this.logout
                : this.login
            }
            user={this.state.user}
   />`

If I take out the prop, wouldn't the login and logout function need to be migrated to the Welcome and NavBar components?

Reply:
That raises a good point of where should that logic be haha.  The login stuff isn't specific to the App class, so it probably shouldn't be there anyway.  You could either extract the logic to a utils file which can be called from either place, or make use of something like redux saga (which is event driven like Redux, but for making async calls).  Setting up Saga now might make complex actions easier later: https://medium.com/@sprocompany/why-you-should-use-redux-saga-b64f6fce5e4a